planemo/workflow_lint.py

```diff
@@ -133,13 +200,37 @@ def _lint_case(path, test_case, lint_context):
             found_valid_expectation = True
         # TODO: validate structure of test expectations
 
+        assertion_definitions = test_case.output_expectations[test_output_id].get("asserts")
+        if not _check_test_assertions(lint_context, assertion_definitions):
+            test_valid = False
+
     if not found_valid_expectation:
         lint_context.warn("Found no valid test expectations for workflow test")
         test_valid = False
 
     return test_valid
 
 
+def _check_test_assertions(lint_context, assertion_definitions):
+    # we are already in Python, not XML, so it is simpler to lint assertions by checking against the
+    # Python functions directly, rather than checking against galaxy.xsd as for tool linting
+    assertions_valid = True
+    if assertion_definitions:
+        for module in asserts.assertion_modules:
+            for function_name in dir(module):
+                if function_name.split('assert_')[-1] in assertion_definitions:
+                    f = module.__dict__[function_name]
```

Comment:
You could check that this is a function with https://docs.python.org/3/library/inspect.html#inspect.isfunction
and inspect the signature with https://docs.python.org/3/library/inspect.html#inspect.signature instead of relying on TypeError. OTOH that's some manual parsing that is maybe not necessary. A healthy middleground could be to run https://docs.python.org/3/library/inspect.html#inspect.getcallargs, so we don't have to execute the test function.

Reply:
> run https://docs.python.org/3/library/inspect.html#inspect.Signature.bind, so we don't have to execute the test function.

That is cool, thanks!